### PR TITLE
Update process lighting

### PR DESCRIPTION
**Pull request by haneslinger**
a brand new process lighting function with 90%+ test coverage! Many of these tests passed on both the old and new function with minor mods. Check out the diff between the two commits.

If you want to see the notebooks I used to create the look up tables, just ask :)

### CODE REVIEW

**Review by kflemin — APPROVED** [review]
looks good!

File: buildingsync_asset_extractor/processor.py [hunk 1]
```diff
@@ -461,12 +465,12 @@ def process_custom_asset(self, asset: dict) -> None:
             # found the name holding the units field
             units_to_export = name_of_units_field[asset['name']]
 
-        custom_assets = {
+        custom_assets: dict[str, Callable[[], list]] = {
```
Comment (nllong) [inline buildingsync_asset_extractor/processor.py:468]: does the other PR (#14) get merged into this first?

File: buildingsync_asset_extractor/processor.py [hunk 2]
```diff
@@ -836,20 +741,14 @@ def format_lighting_results(self, name: str, results: list, units: Optional[str]
...
         # for weighted average, re-find Watts from LPD and LinkedPremises and divide by total sqft
         if has_lpd:
             value = 0.0
             total_sqft = 0
             for r in results:
-                value += r['lpd'] * r['sqft']
-                total_sqft += r['sqft']
+                value += r.lpd * r.sqft
+                total_sqft += r.sqft
             if value > 0:
```
Comment (nllong) [inline buildingsync_asset_extractor/processor.py:752]: catch divide by zero

```suggestion
            if value > 0 and total_sqft != 0:
```

File: tests/lighting_processing/test_lighting_processing.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
+"""
+*********************************************************************************************************
+:copyright (c) BuildingSync®, Copyright (c) 2015-2022, Alliance for Sustainable Energy, LLC,
```
Comment (nllong) [inline tests/lighting_processing/test_lighting_processing.py:3]: 2023

File: tests/lighting_processing/test_lighting_processing.py [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+        # get lighting_systems and clear it
+        lighting_systems_path = "/BuildingSync/Facilities/Facility/Systems/LightingSystems"
+        lighting_systems = self.bp.xp(self.bp.doc, lighting_systems_path)[0]
+        for e in lighting_systems:
+            lighting_systems.remove(e)
+
+        # add method 1 lighting system
+        method_1_ls = etree.XML('''
+            <acc:LightingSystem xmlns:acc="http://buildingsync.net/schemas/bedes-auc/2019">
+                <acc:InstalledPower> 1.0 </acc:InstalledPower>
```
Comment (nllong) [inline tests/lighting_processing/test_lighting_processing.py:109]: spacing around the number looks strange--can you remove the extra spaces, unless there is a clear reason not to.

**Review by nllong — APPROVED** [review]
looks great... nice cleanup and new functionality. 

Few tweaks requested...

File: buildingsync_asset_extractor/lighting_processing/building_type_to_lpd.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+from typing import Optional
+
+
+@dataclass
+class BuildingTypeLPD:
+    building_type: str
+    lpd_by_year: dict[int, Optional[float]]
+
+
+building_type_to_lpd = [
```
[the remark this answers is not in the document]
Comment (haneslinger) [reply buildingsync_asset_extractor/lighting_processing/building_type_to_lpd.py:11]: It's indeed W/ft2 according to the [google sheet ](https://docs.google.com/spreadsheets/d/1ahrr7Q60orZTSrSh0oCb5DwIF1QHMwhQTVM4WPDCsbw/edit#gid=0)

**Pull request merged**